syspage.h

```diff
@@ -19,112 +19,130 @@
 #include <hal/hal.h>
 
 
-#define SIZE_APP_NAME 31
-#define SIZE_MAP_NAME 7
+/* syspage_addProg bitflags */
+enum { flagSyspageExec = 0x01 };
 
 
-/* TODO: Make it compatible with Phoenix-RTOS kernel;
- *       take into account map's attributes while data is written to them */
 enum { mAttrRead = 0x01, mAttrWrite = 0x02, mAttrExec = 0x04, mAttrShareable = 0x08,
 	   mAttrCacheable = 0x10, mAttrBufferable = 0x20 };
 
 
-/* syspage_addProg bitflags */
-enum { flagSyspageExec = 0x01 };
+#pragma pack(push, 1)
 
+typedef struct _syspage_seg_t {
+	struct _syspage_seg_t *next, *prev;
 
-/* syspage_validateAddrMap type of validation bitflags */
-enum { flagValidateTop = 0x01, flagValidateMap = 0x02, flagValidateAttr = 0x04 };
+	addr_t addr;
+	size_t fileSz;
+	size_t memSz;
 
+	u32 flags;
+	u32 align;
+} syspage_seg_t;
 
```

Comment:
After discussion with Hubert I agree that there is no necessity to use two syspage program definitions. If simpler definition fulfills our requirements definition based on segments can be removed.